internal/jet/lateral_table.go

```diff
@@ -0,0 +1,41 @@
+package jet
+
+// LateralTable is interface for Lateral sub-queries
+type LateralTable interface {
```

Comment:
This type already exist -`SelectTable`. No need to add a new one.

Reply:
As LATERAL are a especial kind of subquery that produces a derived table, I wanted to use it's own interface so maybe in the future we can add some constraints, like the accessing variables are not in order etc...

Reply:
We could add it later, if there is a need for new interface.